src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -830,6 +831,38 @@ public function isAbstract()
     }
 
     /**
+     * Whether this definition is deprecated, that means it should not be called
+     * anymore.
+     *
+     * @param bool $boolean
+     *
+     * @return Definition the current instance
+     *
+     * @todo Support a suggested definition instead ?
+     *
+     * @api
+     */
+    public function setDeprecated($boolean = true)
+    {
+        $this->deprecated = (bool) $boolean;
```

Comment:
I'm not quite sure we should remove that, as this is purely a boolean flag (it is deprecated / it is not). Look at how it is handled for the `abstract` property...

I can change this (see @nicolas-grekas' comment on the message and all) to be more flexible ?